query/stats.go

```diff
@@ -57,97 +57,76 @@ func (s *Stat) reset() *Stat {
 
 // statGroup collects simple streaming statistics.
 type statGroup struct {
-	min    float64
-	max    float64
-	mean   float64
+	latencyHDRHistogram *hdrhistogram.Histogram
 	sum    float64
-	values []float64
-
-	// used for stddev calculations
-	m      float64
-	s      float64
-	stdDev float64
-
 	count int64
 }
 
 // newStatGroup returns a new StatGroup with an initial size
 func newStatGroup(size uint64) *statGroup {
+	// This latency Histogram could be used to track and analyze the counts of
+	// observed integer values between 0 us and 30000000 us ( 30 secs )
+	// while maintaining a value precision of 4 significant digits across that range,
+	// translating to a value resolution of :
+	//   - 1 microsecond up to 10 milliseconds,
+	//   - 100 microsecond (or better) from 10 milliseconds up to 10 seconds,
+	//   - 300 microsecond (or better) from 10 seconds up to 30 seconds,
+	lH := hdrhistogram.New(1, 30000000, 4)
 	return &statGroup{
-		values: make([]float64, size),
 		count:  0,
-	}
-}
-
-// median returns the median value of the StatGroup
-func (s *statGroup) median() float64 {
-	sort.Float64s(s.values[:s.count])
-	if s.count == 0 {
-		return 0
-	} else if s.count%2 == 0 {
-		idx := s.count / 2
-		return (s.values[idx] + s.values[idx-1]) / 2.0
-	} else {
-		return s.values[s.count/2]
+		latencyHDRHistogram: lH,
 	}
 }
 
 // push updates a StatGroup with a new value.
 func (s *statGroup) push(n float64) {
-	if s.count == 0 {
-		s.min = n
-		s.max = n
-		s.mean = n
-		s.count = 1
-		s.sum = n
-
-		s.m = n
-		s.s = 0.0
-		s.stdDev = 0.0
-		if len(s.values) > 0 {
-			s.values[0] = n
-		} else {
-			s.values = append(s.values, n)
-		}
-		return
-	}
-
-	if n < s.min {
-		s.min = n
-	}
-	if n > s.max {
-		s.max = n
-	}
-
+	s.latencyHDRHistogram.RecordValue(int64(n * 10e2))
```

Comment:
nit: can this be written as `1e3`, the 10 throws me off since it makes the exponent off by 1 from what i'm used to.

Also since its used so often, probably should be a `const`, like `precisionFactor`?